resteasy-core/src/main/java/org/jboss/resteasy/plugins/providers/ProviderHelper.java

```diff
@@ -160,7 +162,12 @@ public static void writeTo(final InputStream in, final OutputStream out) throws
     */
    public static CompletionStage<Void> writeTo(final InputStream in, final AsyncOutputStream out)
    {
-      final byte[] buf = new byte[2048];
+      final Configuration config = ConfigurationFactory.getInstance().getConfiguration();

```

Comment:
I wonder if we have to read this every time?  If not, I'd prefer to have a member variable that holds this size.  Going the static route is another option.  Do you happen to know what Resteasy's pattern for this is?  Do they tend to respect runtime config changes?